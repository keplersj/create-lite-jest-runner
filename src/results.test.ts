describe("Create Lite Jest Runner Result Creation Helpers", () => {
  describe("Passing Test Helper", () => {
    it("creates a TestResult object", async () => {
      const { pass } = await import("./results");

      expect(
        pass({
          start: 1000,
          end: 2000,
          test: {
            path: "/testPath",
          },
        })
      ).toMatchSnapshot();
    });
  });

  describe("Failing Test Helper", () => {
    it("creates a TestResult object", async () => {
      const { fail } = await import("./results");

      expect(
        fail({
          start: 1000,
          end: 2000,
          test: {
            path: "/testPath",
            errorMessage: "This failed because you're a bad programmer.",
          },
        })
      ).toMatchSnapshot();
    });
  });
});
