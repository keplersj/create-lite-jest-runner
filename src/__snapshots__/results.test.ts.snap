// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Lite Jest Runner Result Creation Helpers Failing Test Helper creates a TestResult object 1`] = `
Object {
  "console": undefined,
  "coverage": undefined,
  "displayName": undefined,
  "failureMessage": "This failed because you're a bad programmer.",
  "leaks": false,
  "memoryUsage": undefined,
  "numFailingTests": 1,
  "numPassingTests": 0,
  "numPendingTests": 0,
  "numTodoTests": 0,
  "openHandles": Array [],
  "perfStats": Object {
    "end": 2000,
    "runtime": 1000,
    "slow": false,
    "start": 1000,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "uncheckedKeys": Array [],
    "unmatched": 0,
    "updated": 0,
  },
  "testExecError": undefined,
  "testFilePath": "/testPath",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "failureDetails": Array [],
      "failureMessages": Array [
        "This failed because you're a bad programmer.",
      ],
      "fullName": "",
      "numPassingAsserts": 0,
      "status": "failed",
      "title": "",
    },
  ],
  "v8Coverage": undefined,
}
`;

exports[`Create Lite Jest Runner Result Creation Helpers Passing Test Helper creates a TestResult object 1`] = `
Object {
  "console": undefined,
  "coverage": undefined,
  "displayName": undefined,
  "failureMessage": undefined,
  "leaks": false,
  "memoryUsage": undefined,
  "numFailingTests": 0,
  "numPassingTests": 1,
  "numPendingTests": 0,
  "numTodoTests": 0,
  "openHandles": Array [],
  "perfStats": Object {
    "end": 2000,
    "runtime": 1000,
    "slow": false,
    "start": 1000,
  },
  "skipped": false,
  "snapshot": Object {
    "added": 0,
    "fileDeleted": false,
    "matched": 0,
    "unchecked": 0,
    "uncheckedKeys": Array [],
    "unmatched": 0,
    "updated": 0,
  },
  "testExecError": undefined,
  "testFilePath": "/testPath",
  "testResults": Array [
    Object {
      "ancestorTitles": Array [],
      "failureDetails": Array [],
      "failureMessages": Array [],
      "fullName": "",
      "numPassingAsserts": 1,
      "status": "passed",
      "title": "",
    },
  ],
  "v8Coverage": undefined,
}
`;
